tools/chplcheck/src/rules.py

```diff
@@ -201,33 +201,46 @@ def DoKeywordAndBlock(
 
         check = node.block_style() != "unnecessary"
         if not check:
-            lines = chapel.get_file_lines(context, node)
-
-            if isinstance(node, Loop):
-                header_loc = node.header_location()
-                if isinstance(node, IndexableLoop) and node.with_clause():
-                    with_ = node.with_clause()
-                    assert with_ is not None
-                    header_loc = header_loc + with_.location()
-            else:
-                header_loc = node.block_header()
-            body_loc = node.curly_braces_location()
-            if header_loc is None or body_loc is None:
-                return check
+            return BasicRuleResult(node, ignorable=True)
+        return check
+
+    @driver.fixit(DoKeywordAndBlock)
+    def RemoveUnnecessaryDo(context: Context, result: BasicRuleResult):
+        """
+        Remove the redundant 'do' keyword before a curly brace '{'.
+        """
+        node = result.node
+        lines = chapel.get_file_lines(context, node)
 
-            header_text = range_to_text(header_loc, lines)
-            body_text = range_to_text(body_loc, lines)
+        if isinstance(node, Loop):
+            header_loc = node.header_location()
+            if isinstance(node, IndexableLoop) and node.with_clause():
+                with_ = node.with_clause()
+                assert with_ is not None
+                header_loc = header_loc + with_.location()
+        else:
+            header_loc = node.block_header()
+        body_loc = node.curly_braces_location()
+        if header_loc is None or body_loc is None:
+            return None
 
-            sep = " "
-            if header_loc.end()[0] != body_loc.start()[0]:
-                indent = " " * (body_loc.start()[1] - 1)
-                sep = "\n" + indent
+        header_text = range_to_text(header_loc, lines)
+        body_text = range_to_text(body_loc, lines)
 
-            new_text = header_text + sep + body_text
-            fixit = Fixit.build(Edit.build(node.location(), new_text))
-            return BasicRuleResult(node, ignorable=True, fixits=[fixit])
+        sep = " "
+        if header_loc.end()[0] != body_loc.start()[0]:
+            indent = " " * (body_loc.start()[1] - 1)
+            sep = "\n" + indent
 
-        return check
+        new_text = header_text + sep + body_text
+        fixit = Fixit.build(Edit.build(node.location(), new_text))
+        return fixit
+
+    def _RedundantParenthesesSubject(node: AstNode):
+        if isinstance(node, (DoWhile, While, Conditional)):
+            return node.condition()
+        elif isinstance(node, IndexableLoop):
+            return node.index()
```

Comment:
Since you have another PR that renames this rule, this function should be renamed. I don't think there will be merge-conflicts, but the names should match

Reply:
Rebased and renamed this and the fixit.